frontend/src/components/worksheets/items/SchemaItem/SchemaItem.js

```diff
@@ -128,13 +131,29 @@ class SchemaItem extends React.Component<{
         const { rows } = this.state;
         let copiedRows = [...rows];
         // replace new line with space, remove single quotes since we use that to quote fields with space when saving
-        copiedRows.splice(idx, 1, {
-            ...rows[idx],
-            [key]: e.target.value.replace(/\n/g, ' ').replace("'", ''),
-        });
+        copiedRows.splice(
+            idx,
+            1,
+            this.autofillTypes({
+                ...rows[idx],
+                [key]: e.target.value.replace(/\n/g, ' ').replace("'", ''),
+            }),
+        );
         this.setState({ rows: [...copiedRows] });
     };
 
+    autofillTypes = (row) => {
+        // Do not fill if how to render is defined.
```

Comment:
I don't understand this comment.

Can you also explain in the comment what `null` means? does it mean that it needs to be autofilled?

Reply:
Yeah `null` means the type is never been auto-filled so we can go ahead and auto-fill it.